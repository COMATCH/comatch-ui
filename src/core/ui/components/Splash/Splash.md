
## Splash

From [`Splash.js`](Splash.js)



prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` | `null` | :x: | 
**id** | `String` | `null` | :x: | 
**inline** | `Boolean` | `false` | :x: | 
**orientation** | `Enum('column','column-reverse','row','row-reverse')` | `'column'` | :x: | 
**primary** | `Boolean` | `false` | :x: | 
**secondary** | `Boolean` | `false` | :x: | 
**subtitle** | `ReactNode` | `null` | :x: | 
**title** | `ReactNode` | `null` | :x: | 



