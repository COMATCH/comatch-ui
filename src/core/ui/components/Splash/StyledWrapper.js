import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { palette, typography } from '../../../styles/variables';

function generateSubTitleStyling() {
    return `
        font-size: ${typography.fontSm};
        line-height: ${typography.lineHeightSm};
    `;
}

function generateTitleStyling() {
    return `
        font-size: ${typography.fontXl};
        line-height: ${typography.lineHeightXl};
    `;
}

const StyledSubtitle = styled.div`${generateSubTitleStyling}`;
const StyledTitle = styled.div`${generateTitleStyling}`;

function generateStyling({ inline, orientation, primary, secondary }) {
    let color = palette.black;
    const flexDirection = [
        'column',
        'column-reverse',
        'row',
        'row-reverse',
    ].includes(orientation)
        ? orientation
        : 'column';

    if (primary) {
        color = palette.primary;
    } else if (secondary) {
        color = palette.gray;
    }

    return `
        color: ${color};
        display: ${inline ? 'inline-flex' : 'flex'};
        flex-direction: ${flexDirection};

        ${StyledSubtitle}, ${StyledTitle} {
            color: ${color};
        }
    `;
}

const StyledWrapper = styled(forwardRef(({ inline, orientation, primary, secondary, ...rest }, ref) => <div {...rest} ref={ref} />))`
    align-items: center;
    ${generateStyling}
`;

export { StyledSubtitle, StyledTitle, StyledWrapper };
