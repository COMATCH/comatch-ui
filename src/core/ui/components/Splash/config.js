import { typography } from '../../../styles/variables';

const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
};

const SIZE = {
    XS: 'xs',
    SM: 'sm',
    MD: 'md',
    LG: 'lg',
    XL: 'xl',
    XXL: 'xxl',
};

const SIZE_TO_PIXELS = {
    xs: {
        title: `font-size: calc(${typography.fontXs} + 6px); line-height: calc(${typography.lineHeightXs} + 6px);`,
        subtitle: `font-size: ${typography.fontXs}; line-height: ${typography.fontXs};`,
    },
    sm: {
        title: `font-size: calc(${typography.fontSm} + 6px); line-height: calc(${typography.lineHeightSm} + 6px);`,
        subtitle: `font-size: ${typography.fontSm}; line-height: ${typography.fontSm};`,
    },
    md: {
        title: `font-size: calc(${typography.fontMd} + 6px); line-height: calc(${typography.lineHeightMd} + 6px);`,
        subtitle: `font-size: ${typography.fontMd}; line-height: ${typography.fontMd};`,
    },
    lg: {
        title: `font-size: calc(${typography.fontLg} + 6px); line-height: calc(${typography.lineHeightLg} + 6px);`,
        subtitle: `font-size: ${typography.fontLg}; line-height: ${typography.fontLg};`,
    },
    xl: {
        title: `font-size: calc(${typography.fontXl} + 6px); line-height: calc(${typography.lineHeightXl} + 6px);`,
        subtitle: `font-size: ${typography.fontXl}; line-height: ${typography.fontXl};`,
    },
    xxl: {
        title: `font-size: calc(${typography.fontXxl} + 6px); line-height: calc(${typography.lineHeightXxl} + 6px);`,
        subtitle: `font-size: ${typography.fontXxl}; line-height: ${typography.fontXxl};`,
    },
};

export { ORIENTATION, SIZE, SIZE_TO_PIXELS };
