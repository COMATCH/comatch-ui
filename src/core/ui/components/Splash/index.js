export { ORIENTATION, SIZE } from './config';
export * from './Splash';
