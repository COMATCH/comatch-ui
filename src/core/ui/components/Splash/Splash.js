import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { StyledSubtitle, StyledTitle, StyledWrapper } from './StyledWrapper';

const propTypes = {
    className: PropTypes.string,
    id: PropTypes.string,
    inline: PropTypes.bool,
    orientation: PropTypes.oneOf(['column', 'column-reverse', 'row', 'row-reverse']),
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    subtitle: PropTypes.node,
    title: PropTypes.node,
};

const defaultProps = {
    className: null,
    id: null,
    inline: false,
    orientation: 'column',
    primary: false,
    secondary: false,
    subtitle: null,
    title: null,
};

const Splash = ({
    className,
    id,
    inline,
    orientation,
    primary,
    secondary,
    subtitle,
    title,
}) => (!!title || !!subtitle) && (
    <StyledWrapper
        {...(id && { id })}
        className={classnames('Splash', className)}
        inline={inline}
        orientation={orientation}
        primary={primary}
        secondary={secondary}
    >
        {!!title && <StyledTitle>{title}</StyledTitle>}
        {!!subtitle && <StyledSubtitle>{subtitle}</StyledSubtitle>}
    </StyledWrapper>
);

Splash.propTypes = propTypes;
Splash.defaultProps = defaultProps;

export { Splash };
