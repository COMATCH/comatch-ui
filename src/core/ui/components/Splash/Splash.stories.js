import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { Splash } from './Splash';

storiesOf('Splash', module)
    .add('with Title', () => (
        <>
            <Splash title="Hello world" />
            <Highlight className="html">{'<Splash title="Hello world" />'}</Highlight>
        </>
    ))
    .add('with Subtitle', () => (
        <>
            <Splash subtitle="Hello world" />
            <Highlight className="html">{'<Splash subtitle="Hello world" />'}</Highlight>
        </>
    ));
