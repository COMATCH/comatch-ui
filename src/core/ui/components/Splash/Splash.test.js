import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Splash } from './Splash';
import '../../../config/tests/setup';

describe('Splash', () => {
    it('should render the default Splash correctly', () => {
        const splash = renderer.create(<Splash />).toJSON();
        expect(splash).toMatchSnapshot();
    });

    it('should render a Splash (with Title) correctly', () => {
        const splash = renderer.create(<Splash title="Title" />).toJSON();
        expect(splash).toMatchSnapshot();
    });

    it('should render a Splash (with Subtitle) correctly', () => {
        const splash = renderer.create(<Splash subtitle="Subtitle" />).toJSON();
        expect(splash).toMatchSnapshot();
    });

    it('should have the correct default props', () => {
        const splash = mount(<Splash />);
        expect(splash.prop('className')).toEqual(null);
        expect(splash.prop('id')).toEqual(null);
        expect(splash.prop('inline')).toEqual(false);
        expect(splash.prop('primary')).toEqual(false);
        expect(splash.prop('secondary')).toEqual(false);
        expect(splash.prop('subtitle')).toEqual(null);
        expect(splash.prop('title')).toEqual(null);
    });

    it('should have the correct passed props', () => {
        const splash = mount((
            <Splash
                className="className"
                id="id"
                inline
                primary
                secondary
                subtitle={<h2>Subtitle</h2>}
                title={<h1>Title</h1>}
            />
        ));

        expect(splash.prop('className')).toEqual('className');
        expect(splash.prop('id')).toEqual('id');
        expect(splash.prop('inline')).toEqual(true);
        expect(splash.prop('primary')).toEqual(true);
        expect(splash.prop('secondary')).toEqual(true);
        expect(splash.prop('subtitle')).toEqual(<h2>Subtitle</h2>);
        expect(splash.prop('title')).toEqual(<h1>Title</h1>);
    });
});
