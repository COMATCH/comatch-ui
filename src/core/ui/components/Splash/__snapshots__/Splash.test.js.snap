// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should render a Splash (with Subtitle) correctly 1`] = `
<div
  className="Splash sc-htpNat edLUgX"
>
  <div
    className="sc-bdVaJa hjJLsw"
  >
    Subtitle
  </div>
</div>
`;

exports[`Splash should render a Splash (with Title) correctly 1`] = `
<div
  className="Splash sc-htpNat edLUgX"
>
  <div
    className="sc-bwzfXH fLSkyy"
  >
    Title
  </div>
</div>
`;

exports[`Splash should render the default Splash correctly 1`] = `null`;
